packages/diagram-ui/src/views/canvas/editableCanvasContentView.ts

```diff
@@ -0,0 +1,136 @@
+import { inject, injectable } from "inversify";
+import { VNode } from "snabbdom";
+import { svg } from "sprotty";
+import { SCanvasElement } from "../../model/canvas/sCanvasElement.js";
+import { CreateConnectionData } from "../../features/create-connection/createConnectionData.js";
+import { LineEngine } from "@hylimo/diagram-common";
+import { SCanvasPoint } from "../../model/canvas/sCanvasPoint.js";
+import { toSVG } from "transformation-matrix";
+import { SCanvasConnection } from "../../model/canvas/sCanvasConnection.js";
+import { TYPES } from "../../features/types.js";
+import { TransactionStateProvider } from "../../features/transaction/transactionStateProvider.js";
+
+/**
+ * Base class for CanvasElementView and CanvasConnectionView
+ */
+@injectable()
+export abstract class EditableCanvasContentView {
+    /**
+     * The transaction state provider
+     */
+    @inject(TYPES.TransactionStateProvider) protected transactionStateProvider!: TransactionStateProvider;
+
+    /**
+     * Renders the create connection preview
+     *
+     * @param model the SRoot model
+     * @returns the rendered create connection preview
+     */
+    protected renderCreateConnection(model: Readonly<SCanvasElement | SCanvasConnection>): VNode | undefined {
+        if (
+            model.createConnectionProvider?.isVisible !== true ||
+            model.selected ||
```

Comment:
Why not when the model has been selected?

Reply:
a) because the displayed outline overlaps with the displayed bounding box of the element in a shitty way
b) because on selected elements, shift + drag is already used for axis-aligned movement (and you don't want to accidentally start creating a line in this case)
c) on selected connections, shift + click is already used to insert a new point
d) we may want to give shift a meaning in the future for resizing elements (e.g. keep aspect ratio)